tasks/firewalld.yml

```diff
@@ -23,3 +23,27 @@
     state: present
     use: "{{ (__firewall_is_ostree | d(false)) |
       ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
+  register: firewall_package_result
+
+- name: Handle reboot for transactional update systems
+  when:
+    - system_is_transactional | d(false)
+    - firewall_package_result is changed
+  block:
+    - name: Notify user that reboot is needed to apply changes
+      debug:
+        msg: >
+          Reboot required to apply changes due to transactional updates.
+
+    - name: Fail if reboot is needed and not set
```

Comment:
In other system roles, the role does not fail, but print a message that reboot is required and sets`storage_reboot_required: true` fact. See https://github.com/linux-system-roles/bootloader/blob/main/tasks/reboot.yml
However, here the role won't be able to proceed without the reboot. So I guess this is fine.